packages/errors/src/codes.ts

```diff
@@ -201,6 +201,7 @@ export const SOLANA_ERROR__TRANSACTION__INVALID_NONCE_TRANSACTION_FIRST_INSTRUCT
     5663014 as const;
 export const SOLANA_ERROR__TRANSACTION__ADDRESSES_CANNOT_SIGN_TRANSACTION = 5663015 as const;
 export const SOLANA_ERROR__TRANSACTION__EMPTY__SIGNATURES_CANNOT_BE_ENCODED = 5663016 as const;
+export const SOLANA_ERROR__TRANSACTION__SIGNATURES_LENGTH_NUM_SIGNERS_MISMATCH = 5663017 as const;
```

Comment:
Agreed, this is hard to understand. What is happening here is, when decoding a transaction, we identified that the decoded `signatures` and the signature requirements encoded in the `messageBytes` do not match. Therefore, what do we think of something like:

```suggestion
export const SOLANA_ERROR__TRANSACTION__MESSAGE_SIGNATURES_MISMATCH = 5663017 as const;
```

And a message like:

> 'The transaction message expected the transaction to have $signaturesLength signatures, got $signerAddressesLength.',

Reply:
That LGTM, will update :)

Reply:
Technically this could be conflated with a situation where the transaction gets the right number of signatures, but one or more is from the wrong key. We might not do a verify step now (or do we?), but we may in the future.